File: StrategyPattern_C++/ConcreteWithdraw3.cpp
#include "ConcreteWithdraw3.h"
#include "SavingAccount.h"


ConcreteWithdraw3::ConcreteWithdraw3()
{
}

ConcreteWithdraw3::~ConcreteWithdraw3()
{
}

int ConcreteWithdraw3::withdraw(Account* acc, int money)
{
	int balance = acc->getBalance();
	Date currDay;
	currDay.year = 2016;
	currDay.month = 10;
	currDay.date = 10;

	if (currDay.year < ((SavingAccount*)acc)->getEndDay().year)
	{
		cout << "Can't withdraw before Maturity date !" << endl;
	}
	else
	{
		if (balance >= money)
		{
			balance -= money;
			cout << "Withdraw Money : " << money << ", Balance : " << balance << endl;
		}
		else
		{
			cout << "Withdraw Money is more than balance!" << endl;
		}
	}

	return balance;
}

File: StrategyPattern_Jave/bank/ConcreteWithdraw3.java
package bank;

import java.util.Date;

public class ConcreteWithdraw3 implements WithdrawStrategy
{
	public void withdraw(Account acc, int money)
	{
		int balance = acc.getBalance();
		Date endDay = ((SavingAccount)acc).getEndDay();
		Date currDay = new Date();
		
		if(currDay.before(endDay))
		{
			System.out.println("Can't withdraw before Maturity date !");
		}
		else
		{
			if(balance >= money)
			{
				balance -= money;
				acc.setBalance(balance);
				System.out.println("Withdraw Money : " + money + ", Balance : " + balance);
			}
			else
			{
				System.out.println("Withdraw Money is more than balance!");
			}
		}
	}
}
